### Scraping the Reddit data 

Initial Steps 
1) Creation of Reddit API - https://www.reddit.com/prefs/apps
2) Make a note of Secret and the Id for the app
3) Install PRAW (Python Reddit API Wrapper) -> Acts like a wrapper for reddit, 
   Navigate - https://praw.readthedocs.io/en/stable/

Good to go 

In [40]:
import praw # Wrapper
import pandas
import matplotlib
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer

In [39]:
nltk.download('vader_lexicon',quiet = True)

True

In [3]:
user = "anonymous"
 
reddit = praw.Reddit(
    client_id = "",
    client_secret = "",
    user_agent= user
    
)

In [31]:
getheadlines = set()
for content in reddit.subreddit('conspiracy').hot(limit=None):
    
    getheadlines.add(content.title)
    # print(content.id)
    # print(content.title)
    # print(content.author)
    
print(len(getheadlines))

915


Inference: 915 Submissions are there in conspiracy community under hot category. 

Lot of features we can access by using subreddit.

Navigate - https://praw.readthedocs.io/en/stable/code_overview/models/subreddit.html#praw.models.Subreddit

In [34]:
redditdf = pandas.DataFrame(getheadlines)
redditdf.head()

0
0  My 3 yo daughter keeps repeating this for no r...
1   Tachyons may be behind precognition/clairvoyance
2                  Lady Diana - Murdered or Accident
3                                    Just my 2 cents
4                Bethel park school shooting threats

Inference: Converted the reddit data (headlines) into a dataframe for further analysis.

In [35]:
redditdf.to_csv("redditdata.csv",encoding='utf-8',index=False,header=False)

Inference: Saved the reddit data into a csv file.

Sentiment Analysis on the Data

In [47]:
analyse = SentimentIntensityAnalyzer()
outcome = []

for row in getheadlines:
    sentiment = analyse.polarity_scores(row)
    sentiment['headlines'] = row
    outcome.append(sentiment)
print(outcome[:1])

[{'neg': 0.258, 'neu': 0.742, 'pos': 0.0, 'compound': -0.4184, 'headlines': 'My 3 yo daughter keeps repeating this for no reason!!'}]
